### PR TITLE
State user crud

### PR DESCRIPTION
**Pull request by aplasket**
# high level what you did:
- created crud functionality for state db and user_state db

## Details:
- add csv file of states
- update rake file
- update state db schema with state_abbv column and state_name
- add index action to state controller and test
- add index and create actions to user_state controller and tests
- add userstate facade for index action in user_state
- add userstate serializer
- add state serializer 
- add Model.destroy_all to seed files for UserCategory, Category, UserState, State, UserBill, Bill
- add dependent: :destroy to State's has_many :user_states, Bill's has_many :user_bills, and Categories's has_many :user_categories

## Overall Checklist:
- [x] tests written for all methods added
- [x] test coverage above 90%
- [x] RSpec running clean
- [x] ran rails s -- features working properly!
- [x] pulled/merged milestone branch before submitting PR

### CODE REVIEW

File: app/controllers/api/v1/user_states_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Api::V1::UserStatesController < ApplicationController
+  def index
+    user_states = UserStateFacade.new.get_states(params[:user_id])
+    render json: StateSerializer.new(user_states)
+  end
+
+  def create
+    u_state = UserState.find_by(user_id: params[:user_id])
+
+    if u_state
```
Comment (jcjurado3) [inline app/controllers/api/v1/user_states_controller.rb:10]: Great use of logic here!

File: lib/tasks/csv_load.rake [hunk 2]
```diff
@@ -8,8 +8,15 @@ namespace :csv_load do
...
+  task :states => :environment do
+    CSV.foreach("./db/data/states.csv", headers: true, header_converters: :symbol) do |row|
+      State.create!(row.to_hash)
+    end
+    puts "States imported."
+  end
+
   task :all do
-    [:categories].each do |task|
+    [:categories, :states].each do |task|
```
Comment (jcjurado3) [inline lib/tasks/csv_load.rake:19]: This is cool, was wondering how to implement this.

**Review by jcjurado3 — APPROVED** [review]
Great code here!

**Review by Jredish11 — APPROVED** [review]

**Review by JCBeldo — COMMENTED** [review]
Looks good!

**Pull request merged**